app/controllers/api/v1/migrations/external_controller.rb

```diff
@@ -33,7 +33,7 @@ def create_role
 
           role = Role.new(name: role_hash[:name], provider: 'greenlight')
 
-          return render_error status: :bad_request unless role.save
+          return render_error(status: :bad_request, errors: role&.errors&.to_a) unless role.save
```

Comment:
Is safe navigation operator  `&.` required here?
This logic should be run after ensuring that the `role` exists.

Reply:
Is it adequate to send raw Activerecord errors?
It can be excessive,

The Activerecord errors can be logged on v3 and they are useful to maintainers not to admins,
There's nothing an admin can do when facing a `Password can't be blank` and that indicates that maintainers made a bug,
Admins can access v3 logs when opening an issue,
I think they need some finer grained specificity for errors not the generic case they have now,

So we can check for unfound records case, invalid data (wrong encryption) and maybe an indication that something went wrong if both cases didn't match and error happened,